### PR TITLE
libsyntax: introduce 'fn is_keyword_ahead(dist, keywords)'.

### PR DESCRIPTION
**Pull request by Centril**
Introduces:
```rust
    /// Returns whether any of the given keywords are `dist` tokens ahead of the current one.
    fn is_keyword_ahead(&self, dist: usize, kws: &[Symbol]) -> bool {
        self.look_ahead(dist, |t| kws.iter().any(|&kw| t.is_keyword(kw)))
    }
```

r? @oli-obk

### CODE REVIEW

File: src/libsyntax/parse/parser.rs [hunk 1]
```diff
@@ -7263,11 +7268,9 @@ impl<'a> Parser<'a> {
         }
         if self.check_keyword(kw::Impl) ||
            self.check_keyword(kw::Unsafe) &&
-                self.look_ahead(1, |t| t.is_keyword(kw::Impl)) ||
-           self.check_keyword(kw::Default) &&
-                self.look_ahead(1, |t| t.is_keyword(kw::Impl)) ||
```
Comment (oli-obk) [inline src/libsyntax/parse/parser.rs]: side-note (pre existing): `default unsafe fn` is eaten by this `if` and will error?

File: src/libsyntax/parse/parser.rs [hunk 2]
```diff
@@ -1087,6 +1087,11 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Returns whether any of the given keywords are `dist` tokens ahead of the current one.
+    fn is_keyword_ahead(&self, dist: usize, kws: &[Symbol]) -> bool {
```
Comment (Centril) [inline src/libsyntax/parse/parser.rs]: We could possibly reduce this to just one keyword... that is one option... :)

File: src/libsyntax/parse/parser.rs (hunk 1 above)
Comment (Centril) [reply src/libsyntax/parse/parser.rs]: Yeah I believe so.

File: src/libsyntax/parse/parser.rs (hunk 2 above)
Comment (petrochenkov) [reply src/libsyntax/parse/parser.rs]: 18 single keyword vs 5 multiple keywords.
\+ `look_ahead` is a single array lookup and a match, more or less, so not much reason to optimize.
Might be reasonable to reduce to one keyword.

**Comment by petrochenkov** [comment]
r=me with or without addressing https://github.com/rust-lang/rust/pull/61321#discussion_r288652722

**Comment by Centril** [comment]
Tricky choice... can't decide one way or the other... but I'll go with what is convenient ;)

@bors r=petrochenkov rollup

**Pull request merged**